### ISSUE TITLE
False positive: Generators suggest wrong missing return type

### ISSUE THREAD

**Issue by kelunik** [kalessil/phpinspectionsea#698]
The following always returns a `\Generator` due to the `yield`, but a return type of `Promise` is suggested.

```php
public function onData(int $clientId, Websocket\Message $msg) {
    if ($msg->isBinary()) {
        return $this->endpoint->sendBinary(yield $msg, $clientId);
    }

    return $this->endpoint->send(yield $msg, $clientId);
}
```

**Comment by kalessil** [comment]
Thank you @kelunik, I'll check the case.

**Comment by kalessil** [comment]
Done, added a workaround (generator is not recognized).
